src/Serialization/main.jl

```diff
@@ -186,6 +190,12 @@ function save_type_params(s::SerializerState, obj::Any, key::Symbol)
   save_type_params(s, obj)
 end
 
+function save_attrs(s::SerializerState, obj::T) where T
+  for attr in attrs_list(s, T)
+    has_attribute(obj, attr) && save_typed_object(s, get_attribute(obj, attr), attr)
+  end
```

Comment:
Wouldn't it make sense to put attributes into their own dict a level deeper (like in https://github.com/oscar-system/Oscar.jl/pull/3980/files#diff-a782bbd162452110d68eee32ed25d1a4269c059bedb762b91fd168937768c8a3R78)? this makes it easier to distinguish attributes from the other serialized parts of the objects, and might reduce some name conflicts

Reply:
I see arguements for both sides, and I think it might be better to just have all properties at the same level.
I also dont think this would lead to name conflicts, I think there would be a deeper issue if there was a property and an attribute with the same name but different values

Reply:
Fine with me. At least, if all types use this helper function, it is consistent between the different types.

Reply:
I pushed a commit, that puts in the attributes into anested layer